tonic/models/vic/grid_params.py

```diff
@@ -665,60 +881,127 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
+        if vegparam_lai and lai_src == FROM_VEGPARAM:
+            varnames.append('LAI')
+        if vegparam_fcan and fcan_src == FROM_VEGPARAM:
+            varnames.append('fcanopy')
+        if vegparam_albedo and alb_src == FROM_VEGPARAM:
+            varnames.append('albedo')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
             new = np.zeros(shape) + FILLVALUE_F
-            new[:-1, :, :] = out_dicts['veg_dict'][var]
-            new[-1, :, :] = 0
+            if extra_class:
+                new[:-1, :, :] = out_dicts['veg_dict'][var]
+                new[-1, :, :] = bare_vegparam[var]
+            else:
+                new[:, :, :] = out_dicts['veg_dict'][var]
+            out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
+
+        varnames = []
+        if blowing_snow:
+            varnames.append('sigma_slope')
+            varnames.append('lag_one')
+            varnames.append('fetch')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
+            new = np.zeros(shape) + FILLVALUE_F
+            if extra_class:
+                new[:-1, :, :] = out_dicts['veg_dict'][var]
+                new[-1, :, :] = bare_vegparam[var]
+            else:
+                new[:, :, :] = out_dicts['veg_dict'][var]
             out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
 
         # Distribute the veglib variables
         # 1st - the 1d vars
         #   double lib_overstory(veg_class) ;  --> (veg_class, nj, ni)
-        for var in ['overstory', 'rarc', 'rmin', 'wind_h', 'RGL', 'rad_atten',
-                    'rad_atten', 'wind_atten', 'trunk_ratio', 'snow_albedo']:
+        varnames = ['overstory', 'rarc', 'rmin', 'wind_h', 'RGL', 'rad_atten',
+                    'rad_atten', 'wind_atten', 'trunk_ratio']
+        for var in varnames:
             lib_var = 'lib_{0}'.format(var)
-            new = np.zeros((nveg_clases, ysize, xsize)) + FILLVALUE_F
-            new[:-1, yi, xi] = veglib_dict[lib_var][:, np.newaxis]
-            new[-1, yi, xi] = 0
+            new = np.zeros((nveg_classes, ysize, xsize)) + FILLVALUE_F
+            if extra_class:
+                new[:-1, yi, xi] = veglib_dict[lib_var][:, np.newaxis]
+                new[-1, yi, xi] = bare_vegparam[var]
+            else:
+                new[:, yi, xi] = veglib_dict[lib_var][:, np.newaxis]
             new[:, ymask, xmask] = fill_val
             out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
 
         # 2nd - the 2d vars
-        for var in ['albedo', 'veg_rough', 'displacement']:
+        varnames = ['veg_rough', 'displacement']
+        if alb_src == FROM_VEGLIB:
+            varnames = ['albedo'] + varnames
+        if veglib_fcan and fcan_src == FROM_VEGLIB:
+            varnames = ['fcanopy'] + varnames
+        if lai_src == FROM_VEGLIB:
+            varnames = ['LAI'] + varnames
```

Comment:
1. The prepending was intentional, although right now I don't recall why I thought that was necessary - at the moment it doesn't seem necessary.
2. OK, will do.